parsl/providers/azure/azure.py

```diff
@@ -0,0 +1,478 @@
+                                                      subscription_id)
+        self.network_client = NetworkManagementClient(credentials,
+                                                      subscription_id)
+
+    def get_credentials(self):
+        """
+        Authenticate to the Azure API
+
+        One of 2 methods are required to authenticate: keyfile, or environment
+        variables. If  keyfile is not set, the following environment
+        variables must be set: `AZURE_CLIENT_ID` (the access key for
+        your azure account),
+        `AZURE_CLIENT_SECRET` (the secret key for your azure account), the
+        `AZURE_TENANT_ID` (the session key for your azure account), and
+        AZURE_SUBSCRIPTION_ID.
+
+        """
+        subscription_id = self.subid
+        credentials = ServicePrincipalCredentials(
+            client_id=self.clientid,
+            secret=self.clientsecret,
+            tenant=self.tenantid)
+        return credentials, subscription_id
+
+    def submit(self,
+               command='sleep 1',
+               blocksize=1,
+               tasks_per_node=1,
+               job_name="parsl.auto"):
+        """
+        Submit the command onto a freshly instantiated Azure VM
+        Submit returns an ID that corresponds to the task that was just submitted.
+        Parameters
+        ----------
+        command : str
+            Command to be invoked on the remote side.
+        blocksize : int
+            Number of blocks requested.
+        tasks_per_node : int (default=1)
+            Number of command invocations to be launched per node
+        job_name : str
+            Prefix for the job name.
+        Returns
+        -------
+        None or str
+            If at capacity, None will be returned. Otherwise, the job identifier will be returned.
+        """
+        self.resource_client.resource_groups.create_or_update(
+            self.group_name, {'location': self.location})
+        self.resources["group"] = self.group_name
+
+        logger.info('Creating NIC')
+        nic = self.create_nic(self.network_client)
+
+        wrapped_cmd = self.launcher(command, tasks_per_node, 1)
+
+        cmd_str = Template(template_string).substitute(jobname=job_name,
+                                                       user_script=wrapped_cmd,
+                                                       linger=str(self.linger).lower(),
+                                                       worker_init=self.worker_init)
+
+        logger.info('Creating Linux Virtual Machine')
+        vm_parameters = self.create_vm_parameters(nic.id,
+                                                  self.vm_reference)
+
+        # Uniqueness strategy from AWS provider
+        job_name = "parsl.auto.{0}".format(time.time())
+
+        async_vm_creation = self.compute_client.\
+            virtual_machines.create_or_update(
+                self.vnet_name, job_name, vm_parameters)
```

Comment:
Using job name in this format of `parsl.auto.{}` is awkwardly here: inside linux, the full hostname is set to `parsl.auto.WHATEVER`, which is generally interpreted as the base hostname being `parsl` and the domain name being `auto.WHATEVER`.

This means that tooling which reports a short hostname always reports `parsl` and it is then hard to tell multiple VMs in the same run apart.

Contrast this with AWS which uses a hostname based on the instance's internal IP address, so the short hostname is reported differently for each VM, something like this: `ip-10-0-4-242`


Put some kind of unique ID into the first bit of the computer name, before the first `.` symbol. time() would do for now, but see #1122